src/osx/carbon/dataobj.cpp

```diff
@@ -63,85 +66,101 @@ wxDataFormat::wxDataFormat( const wxString& rId )
 
 wxDataFormat::wxDataFormat(const wxDataFormat& rFormat)
 {
-    if ( rFormat.m_format )
-        m_format = (NativeFormat) CFStringCreateCopy(NULL, (CFStringRef)rFormat.m_format);
-    else
-        m_format = 0;
+    m_format = wxCFRetain(rFormat.m_format);
     m_type = rFormat.m_type;
-    m_id = rFormat.m_id;
 }
 
-wxDataFormat::wxDataFormat( NativeFormat vFormat )
+wxDataFormat::wxDataFormat(NativeFormat format)
 {
-    m_format = 0;
-    m_type = wxDF_INVALID;
-    SetId( vFormat );
+    SetId(format);
 }
 
 wxDataFormat::~wxDataFormat()
 {
-    if ( m_format != 0 )
-    {
-        CFRelease( (CFStringRef) m_format );
-        m_format = 0;
-    }
+    ClearNativeFormat();
 }
 
-// in order to be correct for 10.3 we restrict to the available types there
-// http://developer.apple.com/qa/qa2005/qa1406.html
-// TODO : Use UTCoreTypes.h constants once we support 10.4+ only
+void wxDataFormat::ClearNativeFormat()
```

Comment:
actually `wxCFStringRef` was only available when pulling in system headers, so I'd have to use `wxCFRef<CFStringRef>` but I'll change this, so that we can use it everywhere